### ISSUE TITLE
[bug] Error : "Unknown Heimdall grant type, please recheck your parameter" on using password grant and client grant

### ISSUE THREAD

**Issue by fahmiegerton** [ezralazuardy/heimdall#2]
**Describe the bug**
I've been tried to using this lib and follow the docs correctly but I used password grant and it throws error. The error is "Unknown Heimdall grant type, please recheck your parameter". I don't know what parameter should I recheck. It doesn't give clear explanation. I would greatly greatful if someone would help me with this issue. Thanks

**To Reproduce**
I'm not quite clear what is this should be about, but I'm debugging via vscode with xdebug, as it can be look at screenshot below. And here's the log

`CRITICAL - 2020-10-03 09:35:04 --> Unknown Heimdall grant type, please recheck your parameter.
#0 D:\wamp64\www\cobacoba\app\Libraries\OAuthServer\OAuthServer.php(54): Heimdall\Heimdall::initializeAuthorizationServer(Object(Heimdall\Config\HeimdallAuthorizationConfig), Object(Heimdall\Config\HeimdallAuthorizationGrant))
#1 D:\wamp64\www\cobacoba\app\Controllers\Rest\Authorization.php(17): App\Libraries\OAuthServer\OAuthServer::createAuthorizationServer()
#2 D:\wamp64\www\cobacoba\vendor\codeigniter4\framework\system\CodeIgniter.php(890): App\Controllers\Rest\Authorization->__construct()
#3 D:\wamp64\www\cobacoba\vendor\codeigniter4\framework\system\CodeIgniter.php(399): CodeIgniter\CodeIgniter->createController()
#4 D:\wamp64\www\cobacoba\vendor\codeigniter4\framework\system\CodeIgniter.php(312): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#5 D:\wamp64\www\cobacoba\public\index.php(45): CodeIgniter\CodeIgniter->run()
#6 {main}`

**Expected behavior**
Know the grant type and generate token

**Screenshots**
![image](https://user-images.githubusercontent.com/5303871/94994243-05df0780-05c9-11eb-9332-e4b018e9e315.png)
![image](https://user-images.githubusercontent.com/5303871/94994119-412d0680-05c8-11eb-8546-4d3fd39ed298.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 Home x64
 - PHP version : 7.3.12 (Using WAMPServer)

**Additional context**
I have no idea

**Comment by andrewrusselrenner** [comment]
I think probably Heimdall currently only handling Authorization Code grant. I've found this when debugging

![image](https://user-images.githubusercontent.com/5303871/95007166-116d1580-063f-11eb-90c3-5da69acebab1.png)

then I've added new case but instead `HeimdallAuthorizationGrant::AuthorizationCode`, I changed it to `HeimdallAuthorizationGrant::PasswordCredentials` but with the same return code. and still no luck, it looking the authorization code on the header after it.
